pkg/manager/manager.go

```diff
@@ -314,6 +314,19 @@ type LeaderElectionRunnable interface {
 	NeedLeaderElection() bool
 }
 
+// WarmupRunnable knows if a Runnable should be a warmup runnable. A warmup runnable is a runnable
+// that should be run when the manager is started, but before the leader election is acquired.
+// Note: Implementing this interface is only useful when LeaderElection is enabled, as the behavior
```

Comment:
nit: Yes and no. Often leader election can be enabled/disabled via command line flags. But at this time the interface is already implemented or not.

Maybe slightly reframe to "when LeaderElection can be enabled"?

Reply:
Addressed in https://github.com/kubernetes-sigs/controller-runtime/pull/3192/commits/6bb4616ba3e651c98015a5382549982074f563ac